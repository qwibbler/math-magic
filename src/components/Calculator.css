#calculator-wrapper {
  width: 29rem;
  text-align: center;
}

#calc-display {
  padding: 1.5rem 1rem;
  text-align: right;
  font-size: 1.5em;
  color: #fff;
  background-color: #332b5c;
  border: 1px solid #f3e7f5;
}

#calc-buttons {
  display: grid;
  grid-template-columns: repeat(4, 1fr);
  background-color: #b0b8db;
}

#\30 {
  grid-column: 1 / 3;
}

input[type="button"] {
  height: 100%;
  width: 100%;
  padding: 1.5rem 1rem;
  background-color: #b0b8db;
  border: 1px solid #f3e7f5;
}

input[type="button"]:hover {
  color: #fff;
  background-color: #4e5572;
}

#calc-buttons > input:last-child,
#calc-buttons > input:nth-child(4n) {
  background-color: #835385;
}

#calc-buttons > input:last-child:hover,
#calc-buttons > input:nth-child(4n):hover {
  background-color: #652e68;
}

@media (max-width: 767px) {
  #calculator-wrapper {
    margin: auto;
    width: 90%;
  }
}
