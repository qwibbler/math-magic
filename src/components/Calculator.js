import React from 'react';
import Button from './Button';
import calculate from '../logic/calculate';

export default class Calculator extends React.Component {
  constructor(props) {
    super(props);
    this.clickHandlerInherit = this.clickHandlerInherit.bind(this);
    this.state = {
      calculatorDataObject: {
        // total: null,
        // next: null,
        // operation: null,
      },
    };
  }

  clickHandlerInherit(buttonName) {
    const { calculatorDataObject } = this.state;
    this.setState(
      {
        calculatorDataObject: calculate(calculatorDataObject, buttonName),
      },
    );
  }

  render() {
    const buttonNames = [
      'AC', '+/-', '%', 'รท',
      '7', '8', '9', 'x',
      '4', '5', '6', '-',
      '1', '2', '3', '+',
      '0', '.', '=',
    ];
    const buttonsArray = [];
    buttonNames.forEach((button) => buttonsArray.push(
      <Button name={button} clickHandlerInherited={this.clickHandlerInherit} />,
    ));
    const { calculatorDataObject } = this.state;

    return (
      <div id="calculator">
        <div id="calc-display">
          <div id="previous">{calculatorDataObject.total}</div>
          <div id="operation">{calculatorDataObject.operation}</div>
          <div id="current">{calculatorDataObject.next}</div>
        </div>
        <div id="calc-buttons">
          {buttonsArray}
        </div>
      </div>
    );
  }
}
