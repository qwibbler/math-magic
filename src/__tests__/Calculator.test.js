import React from 'react';
import renderer from 'react-test-renderer';
import { render, fireEvent } from '@testing-library/react';
import Calculator from '../components/Calculator';

describe('Calculator', () => {
  test('Calculator renders correctly', () => {
    const tree = renderer.create(<Calculator />).toJSON();
    expect(tree).toMatchSnapshot();
  });

  test('Button click correctly', () => {
    render(<Calculator />);
    const button5 = document.getElementById('5');
    fireEvent.click(button5);
    const display = document.getElementById('calc-display');
    expect(display.textContent[2]).toBe('5');
  });
});
