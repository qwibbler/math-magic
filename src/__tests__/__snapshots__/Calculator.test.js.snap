// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Calculator Calculator renders correctly 1`] = `
<div
  id="calculator-wrapper"
>
  <div
    id="calc-display"
  >
     
     
  </div>
  <div
    id="calc-buttons"
  >
    <input
      id="AC"
      onClick={[Function]}
      type="button"
      value="AC"
    />
    <input
      id="+/-"
      onClick={[Function]}
      type="button"
      value="+/-"
    />
    <input
      id="%"
      onClick={[Function]}
      type="button"
      value="%"
    />
    <input
      id="÷"
      onClick={[Function]}
      type="button"
      value="÷"
    />
    <input
      id="7"
      onClick={[Function]}
      type="button"
      value="7"
    />
    <input
      id="8"
      onClick={[Function]}
      type="button"
      value="8"
    />
    <input
      id="9"
      onClick={[Function]}
      type="button"
      value="9"
    />
    <input
      id="x"
      onClick={[Function]}
      type="button"
      value="x"
    />
    <input
      id="4"
      onClick={[Function]}
      type="button"
      value="4"
    />
    <input
      id="5"
      onClick={[Function]}
      type="button"
      value="5"
    />
    <input
      id="6"
      onClick={[Function]}
      type="button"
      value="6"
    />
    <input
      id="-"
      onClick={[Function]}
      type="button"
      value="-"
    />
    <input
      id="1"
      onClick={[Function]}
      type="button"
      value="1"
    />
    <input
      id="2"
      onClick={[Function]}
      type="button"
      value="2"
    />
    <input
      id="3"
      onClick={[Function]}
      type="button"
      value="3"
    />
    <input
      id="+"
      onClick={[Function]}
      type="button"
      value="+"
    />
    <input
      id="0"
      onClick={[Function]}
      type="button"
      value="0"
    />
    <input
      id="."
      onClick={[Function]}
      type="button"
      value="."
    />
    <input
      id="="
      onClick={[Function]}
      type="button"
      value="="
    />
  </div>
</div>
`;
